decentralizediso20022/src/main/java/com/centiglobe/decentralizediso20022/presentation/external/ExtPacsController.java

```diff
@@ -62,16 +55,18 @@ public ResponseEntity handlePacs008(@RequestBody String pacs) throws Unsupported
         AbstractMX mx = AbstractMX.parse(decodedPacs);
         if (mx == null || !mx.getBusinessProcess().equals("pacs"))
             throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The entity was not a valid pacs message.");
-        
+
         BusinessAppHdrV02 header = (BusinessAppHdrV02) mx.getAppHdr();
         try {
-            // Extract path relative to the classpath
-            String truststore = new File(TRUST_STORE).getName();
             int port = Integer.parseInt(PORT);
-            ValidationService.validateHeaderFrom(header, port, truststore, TRUST_PASS);
-            ValidationService.validateHeaderTo(header, port, truststore, TRUST_PASS);
-        } catch (Exception e) {
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The entity had an invalid from or to header.");
+            vs.validateHeader(header, port);
+        } catch (Throwable e) {
+            // Note: Might want check for WebClientResponseException, instead of only
+            // NotFound.
+            if (!(e instanceof WebClientResponseException.NotFound)) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
+                        "The entity had an invalid from or to header.");
+            }

```

Comment:
If I've understood correctly, if `e` is not an instance of `WebClientResponseException.NotFound`, the ISO 20022 message will be forwarded to the bank system even though the header was invalid. Is there any reason why a `ResponseStatusException` is not thrown regardless of how `vs.validateHeader` failed?

Reply:
I need to do some further testing, but if the header is invalid it should not be forwarded. I'm only catching the `WebClientResponseException.NotFound`, no exceptions related to SSL.

The reason I had to do this is that on `localhost` I make a request to the domain's `/` endpoint, which is not configured to anything. Maybe we should not accept the certificate even if it is valid if the webpage returns a 404, or anything else than a 200?

Reply:
The only thing we want to confirm is that the hostname has a valid certificate, even if requesting their `/` results in a 404 or another error code. We shouldn't require the banks to have a valid webpage on their `/` right?

Reply:
That's why I don't throw an error on 404.

Reply:
An why I think we might not want to throw an error on any `WebClientResponseException`.

Reply:
OK,  I'm with you that we should ignore `WebClientResponseException`s. Coudl you change it to that and add a TODO that we should check if there are any other exceptions that we should ignore?

Reply:
I'll do that